code/modules/mob/mob.dm

```diff
@@ -408,8 +406,10 @@
 
 				return //And we're good
 		else //This is not an emote
-			to_chat(src, "<span class='notice'>You can almost hear someone talking.</span>")//The sweet silence of death
-
+			if (speaker && (src.ckey == speaker.ckey) && speaker.isincrit() && speaker.said_last_words) //if user is in crit, if user has said_last_words, and whisperer of the final words IS the user himself)
```

Comment:
`(src.ckey == speaker.ckey)` ; any reason why you're not making this `speaker == src` ?

Reply:
Wasn't sure if `speaker == src` would always pass. But it should be the same really.

Reply:
Works for me